driver-service/src/main/java/com/modsen/driverservice/service/impl/DriverServiceImpl.java

```diff
@@ -0,0 +1,180 @@
+        if (driverRepository.existsByPhone(phone)) {
+            throw new PhoneAlreadyExistException(String.format(ExceptionMessages.DRIVER_WITH_PHONE_ALREADY_EXIST,phone));
+        }
+    }
+
+    private void checkCarNumberExist(String carNum) {
+        if (driverRepository.existsByNumber(carNum)) {
+            throw new CarNumberAlreadyExistException(String.format(ExceptionMessages.DRIVER_WITH_CAR_NUMBER_ALREADY_EXIST, carNum));
+        }
+    }
+
+    public DriverListResponse getAvailableDrivers() {
+        List<Driver> listOfAvailableDrivers = driverRepository.getAllByAvailable(true);
+        List<DriverResponse> listOfAvailable =  listOfAvailableDrivers.stream()
+                .map(this::fromEntityToResponse)
+                .toList();
+        return new DriverListResponse(listOfAvailable);
+    }
+
+    private PageRequest getPageRequest(int page, int size, String orderBy) {
+        if (page < 1 || size < 1) {
+            throw new PaginationParamException(String.format(ExceptionMessages.PAGINATION_FORMAT_EXCEPTION));
+        }
+
+        PageRequest pageRequest;
+        if (orderBy == null) {
+            pageRequest = PageRequest.of(page - 1, size);
+        } else {
+            validateSortingParameter(orderBy);
+            pageRequest = PageRequest.of(page - 1, size, Sort.by(orderBy));
+        }
+
+        return pageRequest;
+    }
+
+    public DriverPageResponse getDriverPage(int page, int size, String orderBy) {
+
+        PageRequest pageRequest = getPageRequest(page, size, orderBy);
+        Page<Driver> driverPage = driverRepository.findAll(pageRequest);
+
+        List<Driver> retrievedDrivers = driverPage.getContent();
+        long total = driverPage.getTotalElements();
+
+        List<DriverResponse> passengers = retrievedDrivers.stream()
+                .map(this::fromEntityToResponse)
+                .toList();
+
+        return DriverPageResponse.builder()
+                .driverList(passengers)
+                .totalPages(page)
+                .totalElements(total)
+                .build();
+    }
+
+    private void validateSortingParameter(String orderBy) {
+        List<String> fieldNames = Arrays.stream(DriverResponse.class.getDeclaredFields())
+                .map(Field::getName)
+                .filter(field -> !field.equals(orderBy))
+                .toList();
+
+        if (fieldNames.isEmpty()) {
+            throw new SortTypeException(ExceptionMessages.INVALID_TYPE_OF_SORT);
+        }
+    }
+
+    private Driver getOrThrow(Long id){
+        return driverRepository.findById(id)
+                .orElseThrow(()-> new DriverNotFoundException(String.format(ExceptionMessages.DRIVER_NOT_FOUND_EXCEPTION,id)));
+    }
+
+    public DriverResponse changeStatus(Long driverId) {
+        Driver driver = getOrThrow(driverId);
+        driver.setAvailable(!driver.isAvailable());
+        driverRepository.save(driver);
+        return fromEntityToResponse(driver);
+    }
```

Comment:
Я бы добавил пустую строку после  `Driver driver = getOrThrow(driverId);`
Так код разграничивается на логические блоки, что увеличивает читаемость
```suggestion
        Driver driver = getOrThrow(driverId);
        
        driver.setAvailable(!driver.isAvailable());
        driverRepository.save(driver);
        return fromEntityToResponse(driver);
    }
```